### PR TITLE
match plugins, add yum config

### PR DESCRIPTION
**Pull request by hopkiw**
support config and related auth options in yum, matching dnf

### CODE REVIEW

**Comment by hopkiw** [comment]
/hold
we will want to merge this after packaging, as it will have packaging-related updates (share the conf file)

File: yum/artifact-registry.py [hunk 1]
```diff
@@ -23,25 +25,46 @@
...
         break  # Stop looking at URLs
 
 
-def _add_headers(repo):
+def _get_creds(conduit):
+  service_account_json = conduit.confString('main', 'service_account_json', '')
+  if service_account_json:
+    return service_account.Credentials.from_service_account_file(
+        service_account_json)
+  service_account_email = conduit.confString('main', 'service_account_email',
```
Comment (illfelder) [inline yum/artifact-registry.py:43]: Please line wrap after the open paren.

File: artifact-registry.conf [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Options for the Google Artifact Registry YUM and DNF plugins.
```
Comment (illfelder) [inline artifact-registry.conf:1]: nit: `s/Google Artifact Registry/Artifact Registry/`

**Comment by hopkiw** [comment]
dnf:
```
$ rpm -qpilv ./dnf-plugin-artifact-registry-dummy-g1.el8.x86_64.rpm 
Name        : dnf-plugin-artifact-registry
Epoch       : 1
Version     : dummy
Release     : g1.el8
Architecture: x86_64
Install Date: (not installed)
Group       : Unspecified
Size        : 16567
License     : ASL 2.0
Signature   : (none)
Source RPM  : dnf-plugin-artifact-registry-dummy-g1.el8.src.rpm
Build Date  : Tue 09 Mar 2021 10:08:05 PM UTC
Build Host  : inst-build-pkg-build-packages-el8-build-package-pj7w0.c.gcp-guest.internal
Relocations : (not relocatable)
URL         : https://cloud.google.com/artifact-registry
Summary     : dnf plugin for Artifact Registry
Description :
Contains a dnf plugin for authenticated access to Artifact Registry repositories.
-rw-r--r--    1 root    root                      509 Mar  9 22:07 /etc/dnf/plugins/artifact-registry.conf
-rw-r--r--    2 root    root                     2243 Mar  9 22:08 /usr/lib/python3.6/site-packages/dnf-plugins/__pycache__/artifact-registry.cpython-36.opt-1.pyc
-rw-r--r--    2 root    root                     2243 Mar  9 22:08 /usr/lib/python3.6/site-packages/dnf-plugins/__pycache__/artifact-registry.cpython-36.pyc
-rw-r--r--    1 root    root                     2457 Mar  9 22:07 /usr/lib/python3.6/site-packages/dnf-plugins/artifact-registry.py
drwxr-xr-x    2 root    root                        0 Mar  9 22:08 /usr/share/doc/dnf-plugin-artifact-registry
-rw-r--r--    1 root    root                    11358 Mar  9 22:07 /usr/share/doc/dnf-plugin-artifact-registry/LICENSE
```
and yum:
```
$ rpm -qpilv ./yum-plugin-artifact-registry-dummy-g1.el7.x86_64.rpm 
Name        : yum-plugin-artifact-registry
Epoch       : 1
Version     : dummy
Release     : g1.el7
Architecture: x86_64
Install Date: (not installed)
Group       : Unspecified
Size        : 16019
License     : ASL 2.0
Signature   : (none)
Source RPM  : yum-plugin-artifact-registry-dummy-g1.el7.src.rpm
Build Date  : Tue 09 Mar 2021 10:03:07 PM UTC
Build Host  : inst-build-pkg-build-packages-el7-build-package-70hrn.c.gcp-guest.internal
Relocations : (not relocatable)
URL         : https://cloud.google.com/artifact-registry
Summary     : Yum plugin for Artifact Registry
Description :
Contains a Yum plugin for authenticated access to Artifact Registry repositories.
-rw-r--r--    1 root    root                      509 Mar  9 22:03 /etc/yum/pluginconf.d/artifact-registry.conf
-rw-r--r--    1 root    root                     2102 Mar  9 22:03 /usr/lib/yum-plugins/artifact-registry.py
-rw-r--r--    2 root    root                     2050 Mar  9 22:03 /usr/lib/yum-plugins/artifact-registry.pyc
-rw-r--r--    2 root    root                     2050 Mar  9 22:03 /usr/lib/yum-plugins/artifact-registry.pyo
drwxr-xr-x    2 root    root                        0 Mar  9 22:03 /usr/share/doc/yum-plugin-artifact-registry-dummy
-rw-r--r--    1 root    root                    11358 Mar  9 22:03 /usr/share/doc/yum-plugin-artifact-registry-dummy/LICENSE
```

**Pull request merged**